### PR TITLE
Removing some unnecessary transitive dependencies from the hbase-client

### PR DESCRIPTION
**Pull request by sduskis**

### CODE REVIEW

File: bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableConfiguration.java [hunk 1]
```diff
@@ -203,6 +199,10 @@ public Builder toBuilder() {
   }
 
   public boolean equals(Object obj) {
-    return EqualsBuilder.reflectionEquals(this, obj);
+    if (obj == null || obj.getClass() != this.getClass()) {
```
Comment (kevinsi4508) [inline bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableConfiguration.java]: obj instanceof CloudBigtableConfiguration?

File: bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableScanConfiguration.java [hunk 2]
```diff
@@ -230,6 +231,13 @@ public Scan getScan() {
   }
 
   @Override
+  public boolean equals(Object obj) {
+    return super.equals(obj)
+        && Objects
+            .equals(serializableScan, ((CloudBigtableScanConfiguration) obj).serializableScan);
```
Comment (kevinsi4508) [inline bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableScanConfiguration.java]: do u want to check if obj is null, similar to CloudBigtableConfiguration.java

File: bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableConfiguration.java (hunk 1 above)
Comment (sduskis) [inline bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableConfiguration.java]: If it's not the same class (i.e. if objs or this is a different subclass), then the objects are not equals.

File: bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableScanConfiguration.java (hunk 2 above)
Comment (sduskis) [inline bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableScanConfiguration.java]: is it worth doing even though super.equals(obj) does it?

Comment (kevinsi4508) [inline bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableScanConfiguration.java]: missed super.equals(obj). That is fine.

**Comment by kevinsi4508** [comment]
LGTM

**Pull request merged**